frodo.js

```diff
@@ -3,130 +3,133 @@
  *
```

Comment:
SCRIPT-I'm a bit concerned about the potential performance impact of this loop. Have you thought about optimizing it?